dashboard/test/models/sections/section_test.rb

```diff
@@ -71,6 +71,59 @@ class SectionTest < ActiveSupport::TestCase
     end
   end
 
+  test 'changing sharing_disabled updates user settings' do
+    student = create :student, sharing_disabled: false
+    section = create :section, sharing_disabled: false
+    assert_creates(Follower) do
+      section.add_student student
+    end
```

Comment:
There are a lot of these `assert_creates(Follower)` that seem extraneous to the test. We already have a [test](https://github.com/code-dot-org/code-dot-org/blob/ea6b02539634f8599954570f694dd9388e3b6fe7/dashboard/test/models/sections/section_test.rb#L134-L141) for `add_student`. I don't think we need to test it here.

I'd prefer to stick with relevant asserts. 

We could even update the `:section` factory to take a set of students so we can do something like `section = create :section, students: [student]` and not even call `add_student`. That currently doesn't work because of the [initialize_with](https://github.com/code-dot-org/code-dot-org/blob/192d3f8f6708b28a8e402f79bd163668e6321d59/dashboard/test/factories/factories.rb#L202), which was [added](https://github.com/code-dot-org/code-dot-org/pull/16371) to ensure the right STI class is used. We'd have to add a transient attribute, `students` to the `section` factory, and then an `after(:create)` block to add them. This is just an idea. I'm not convinced it's a huge improvement:

```ruby
student = create :student
section = create :section
section.add_student
```
vs.
```ruby
student = create :student
section = create :section, students: [student]
```